### PR TITLE
ssbc: Implement workspace resolver in backend

### PR DESCRIPTION
**Pull request by eseliger**
This PR pulls in two pieces of code from src-cli:
- `FindDirectoriesInRepos `
- `findWorkspaces`
and their test suites.
We're actually not doing much magic here, rename a couple of methods to return workspaces and not repoRevs, slightly change the GraphQL schema to expose the new information we have at hand now and let the frontend make use of that data.

### CODE REVIEW

File: enterprise/cmd/frontend/internal/batches/resolvers/batch_spec_workspace.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+var _ graphqlbackend.BatchSpecWorkspaceResolver = &batchSpecWorkspaceResolver{}
+
+func (r *batchSpecWorkspaceResolver) Repository(ctx context.Context) (*graphqlbackend.RepositoryResolver, error) {
+	return graphqlbackend.NewRepositoryResolver(r.store.DB(), r.node.Repo), nil
+}
+
+func (r *batchSpecWorkspaceResolver) Branch(ctx context.Context) (*graphqlbackend.GitRefResolver, error) {
+	repo, _ := r.Repository(ctx)
+	return graphqlbackend.NewGitRefResolver(repo, r.node.Branch, graphqlbackend.GitObjectID(r.node.Commit)), nil
```
Comment (eseliger) [inline enterprise/cmd/frontend/internal/batches/resolvers/batch_spec_workspace.go:25]: todo: validate this points to the correct commit

File: enterprise/internal/batches/service/workspace_resolver_test.go [hunk 2]
```diff
@@ -405,3 +417,270 @@ func mockResolveRevision(t *testing.T, branches map[string]api.CommitID) {
 	}
 	t.Cleanup(func() { git.Mocks.ResolveRevision = nil })
 }
+
+func TestFindWorkspaces(t *testing.T) {
```
Comment (eseliger) [inline enterprise/internal/batches/service/workspace_resolver_test.go:421]: Copied from src-cli and adjusted for this env.

File: enterprise/cmd/frontend/internal/batches/resolvers/batch_spec_workspace.go (hunk 1 above)
Comment (eseliger) [reply enterprise/cmd/frontend/internal/batches/resolvers/batch_spec_workspace.go:25]: I'll validate that now.

Comment (eseliger) [reply enterprise/cmd/frontend/internal/batches/resolvers/batch_spec_workspace.go:25]: Seems to be correct.

**Review by mrnugget — APPROVED** [review]
Reviewed on Zoom :+1:

**Pull request merged**